src/test/java/unit/DataSourceReferenced.java

```diff
@@ -181,6 +183,42 @@ public void resolve() throws IOException {
         "resource", "delete", "--name=" + DATA_SOURCE_RESOURCE_NAME, "--quiet");
   }
 
+  @Test
+  @DisplayName("Test running terra app execute env in a workspace with data source")
+  void appExecuteEnv() throws IOException {
+    workspaceCreator.login();
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + getUserFacingId());
+
+    // `terra app execute env`
+    Result result = TestCommand.runAndGetResultExpectSuccess("app", "execute", "env");
+    String stdOut = result.stdOut;
+    assertDataSourceResourcesEnv(stdOut);
+
+    // `terra resource delete --name=$name`
+    TestCommand.runCommandExpectSuccess(
```

Comment:
Can we remove delete from this test and resolve() test? listDescribeDelete() runs delete because it confirms delete works.

(Workspaces should automatically be cleaned up by @zloery 's janitor work -- @zloery , is that turned on?)

Reply:
yep that's running for `broad` config tests, I think workspaces should be cleaned up after 1 week

Reply:
oh yeah. i added that because the set up was not @beforeeach before. removed.